FrontEndLib/Sound.cpp

```diff
@@ -985,7 +985,9 @@ SOUNDSTREAM* CSound::LoadSongStream(const WSTRING& wstrSongFilepath, UINT mode)
 	WideCharToMultiByte(CP_ACP, 0, wstrSongFilepath.c_str(), -1, sANSI, 512, NULL, NULL);
```

Comment:
I can't recall why we're using WideCharToMultiByte here anymore.  We should replace it.

Not your fault, of course, but I see this function has multiple risks:
https://docs.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-widechartomultibyte

While we're improving management of sANSI from a char[] (see my remark below), we should switch sANSI to a string for Windows usage also and use UnicodeToUTF8 to convert wstrSongFilepath to a UTF8 string that FMOD can use to load files.  (Heh...this reminds me of the string conversion issue with the Metakit file loading problem...)

@EvidentlyCube  (same in catch.cpp, probably, to avoid requiring a Windows-specific function for command line arg conversion.)

Reply:
I can try to make this change.